File: backend/models/attendance.go
package models

import "time"

type Attendance struct {
	ID               uint      `json:"id"`
	TapType          int8      `json:"tap_type"`
	ValidationStatus int8      `json:"validation_status"`
	CreatedAt        time.Time `json:"created_at"`
	RfidID           uint      `json:"rfid_id"`
	LocationID       uint      `json:"location_id"`
}

File: src/types/attendance.ts
export type Attendance = {
    id: number;
    rfid_id: number;
    rfid_tag: string;
    location_id: number;
    location: string;
    tap_type: 1 | 2; // 1 = Masuk, 2 = Keluar
    validation_status: 1 | 2 | 3 | 4 | 5; // 1: Hadir, 2: Terlambat, 3: Izin, 4: Sakit, 5: Bolos/Tidak Hadir
    created_at: string;
}